scripts/test-util.sh

```diff
@@ -28,6 +29,10 @@ process_common_args() {
                 shift
                 test_filter="$1"
                 ;;
+            "--custom-args")
```

Comment:
I think this a kludgy solution. It would be better to "consume" the args from here and then return the residual args from `$@` as something else. This means it can be used for all the scripts not just specific ones.

Reply:
Also doesn't the problem come from this `test-util.sh` script since it does some parsing, which drops anything that hasn't been parsed?
I mean IIRC before the introduction of this script, we could properly use `--` to separate the extra/custom parameters, couldn't we? If I'm correct, don't you think we should try an alternative approach for identifying the hypervisor type, which would let us restore the previous behavior. This way we wouldn't need things like `test_filter` either.

Reply:
@rbradford @sboeuf PR updated. PTAL. I ended up with using an additional separator `--` to specify that a certain set of arguments passed to "dev_cli.sh" will be passed through to the test binary (both `cargo test` and `performance-metrics`).  The following two example works as expected:

1. "`$ ./dev_cli.sh tests --metrics -- -- --report-file /tmp/metrics.json --test-filter latency`"  will do "`performance-metrics --report-file /tmp/metrics.json --test-filter latency`"; 
2. "`$ ./dev_cli.sh tests --integration -- --test-filter "test_serial"  -- --nocapture --test-threads=1`" will do "`cargo test parallel::test_serial -- --nocapture --test-threads=1`".

Please let me me know whether it works for the use cases you have in mind.

Reply:
A side note: I also tried to pass-through any unrecognized arguments directly to the test binary (https://github.com/likebreath/cloud-hypervisor/commit/dcf0506e86236d314bb13ad3a6fe13e55ba0776d), but it does not work well for the arguments that are ambiguous. For instance, the `--test-filter` option is valid for both the test script (parsed by `test-util.sh`) and `performance-metrics`, but not valid for `cargo test --`. 

If this is the solution we want to adopt, we will need to make sure we don't have any ambiguous arguments manually, e.g. we will need to rename `--test-filter` from either `test-util.sh` or `performance-metrics`.

Reply:
I don't think we need the extra patch you propose here.